src/com/amazon/ion/impl/lite/IonValueLite.java

```diff
@@ -962,10 +1039,8 @@ public void makeReadOnly()
         }
     }
 
-    void makeReadOnlyInternal()
-    {
-        clearSymbolIDValues();
-        _isLocked(true);
+    void makeReadOnlyInternal() {
+        clearSymbolIDsIterative(true);
```

Comment:
It's probably worth adding a shortcut for scalars here too, as you did for clearSymbolIDValues.